admin_guide/sdn_troubleshooting.adoc

```diff
@@ -0,0 +1,527 @@
+....
+# ip route get 192.168.122.202
+  192.168.122.202 dev ens3  src 192.168.122.46
+    cache
+....
+** In this case it says it will go out the interface named `ens3` with the source address of 192.168.122.46 and go directly to the target.  Is that what you expected?
+*** If so, use `ip a show dev ens3` to get the interface details and make sure that is the expected interface
+** The other possibility is that it might say:
+....
+# ip route get 192.168.122.202
+  1.2.3.4 via 192.168.122.1 dev ens3  src 192.168.122.46
+....
+*** If so, the via means that it will pass it to that router to route
+appropriately... you will have to make sure that the router is doing
+the right thing with the traffic, and debugging that is beyond the
+scope of this guide.
+* Do you have ethernet link (on both ends)?  (Look for `Link detected: yes` in the output from `ethtool _interface_`)
+* Are your duplex settings, and ethernet speeds right (on both ends)? (Look through the rest of the `ethtool _interface_` information)
+* Are the cables plugged in correctly?  To the correct ports?
+* Are the switches configured correctly?
+
+
+Once you have ascertained that the node to node connectivity is fine,
+we need to look at the SDN configuration on both ends.  Please proceed
+to the <<debugging-local-networking>> section.
+
+
+== Debugging Local Networking
+
+At this point we should have a list of one or more endpoints that you
+can't communicate with, but that have node to node connectivity.  For
+each one, we need to work out what is wrong, but first you need to
+understand how the SDN sets up the networking on a node for the
+different pods.
+
+=== The Interfaces on a Node
+
+These are the interfaces that the OpenShift SDN creates:
+
+* `br0`: The OVS bridge device that containers will be attached to.
+OpenShift SDN also configures a set of non-subnet-specific flow rules
+on this bridge. (The `multitenant` plugin does this immediately; the
+`ovssubnet` plugin waits until the SDN master announces the creation
+of the new node subnet.)
+* `lbr0`: A Linux bridge device, which is configured as Docker's bridge and
+given the cluster subnet gateway address (eg, 10.1.x.1/24).
+* `tun0`: An OVS internal port (port 2 on `br0`). This *also* gets assigned the
+cluster subnet gateway address, and is used for external network
+access. OpenShift SDN configures `netfilter` and routing rules to enable access
+from the cluster subnet to the external network via NAT.
+* `vlinuxbr` and `vovsbr`: Two Linux peer virtual Ethernet interfaces.
+`vlinuxbr` is added to `lbr0`, and `vovsbr` is added to `br0` (port 9), to
+provide connectivity for containers created directly with Docker outside of
+OpenShift.
+* `vxlan0`: The OVS VXLAN device (port 1 on `br0`), which provides access to
+containers on remote nodes.
+* `vethX` (in the main netns): A Linux virtual ethernet peer of `eth0` in the docker netns.  It will be attached to the OVS bridge on one of the ports 3-8 or 10.
```

Comment:
For ovssubnet it's 3 or higher except not 9. for multitenant it's 4 or higher. At any rate, I don't think you need to explain this, since either (a) they won't know the port number of the veth they care about, in which case it doesn't matter what it *might* be, or (b) they will know the port number of the veth they care about, in which case it doesn't matter what it might have been.